### ISSUE TITLE
Reassign deleted user data / fully delete user

### ISSUE THREAD

**Issue by ssuess** [WeblateOrg/weblate#3578]
**PROBLEM**
We have hooked up weblate to our SSO, and it works fine until a user deletes themselves, after which they will be unable to log back in because although their account has been deleted, it is not TRULY deleted, just renamed to deleted*. The unique ID is still in the system and blocks creation/login

**SOLUTION PROPOSAL**
It would seem that an easy fix for this would be to assign all of that user's suggestions/contributions/etc to some other user designated to hold this data and then fully delete the original user. After that, if the SSO were to login again, an account could be recreated because the userID would no longer be in use.

Another solution would be to restore the user upon login with that ID. (and possibly provide a dialog that asks them "This user was previously deleted. Are you sure you want to restore it?"

**Describe alternatives you've considered**
Looked through the docs, but nothing seemed to fit this need.

**Comment by nijel** [comment]
I think that in your case it would make sense to disable account removal.

**Comment by ssuess** [comment]
A. Wouldn't that run counter to GDPR? B. Why shouldn't we (theoretically) be able to fully delete accounts?

**Comment by nijel** [comment]
GDPR doesn't request there is UI for it. Admin would still be able to do this. It just would be done in rare cases.

Reassigning all contributions to other users is problematic. It would make all things from removed users appear as from one, what would make it hard to follow discussions or suggestions. The other reason is that it would be easy to make it wrong (missing moving some content to other user) what would lead to database corruption on actual user removal.

Anyway your case is special. Normally in Weblate accounts are identified by email, which is removed on account removal, what makes it possible for user to register again.

For your use case, you can also just set "is active" flag on the deleted user in the admin interface, and he should be able to login again.

**Comment by ssuess** [comment]
I will look into doing that, thanks for the response!
